src/main/java/shef/servlets/PreferenceServlet.java

```diff
@@ -0,0 +1,127 @@
+      put("halal", Arrays.asList("HALAL", "ALCOHOL-FREE", "ALCOHOL FREE", "NO ALCOHOL", "PORK-FREE", "PORK FREE", "NO PORK"));
+      put("no-dairy", Arrays.asList("DAIRY-FREE", "DAIRY FREE", "NO DAIRY"));
+      put("no-shellfish", Arrays.asList("SHELLFISH-FREE", "SHELLFISH FREE", "NO SHELLFISH"));
+      put("no-seafood", Arrays.asList("SEAFOOD-FREE", "SEAFOOD FREE", "NO SEAFOOD", "FISH-FREE", "FISH FREE", "NO FISH", "SHELLFISH-FREE", "SHELLFISH FREE", "NO SHELLFISH"));
+      put("no-soy", Arrays.asList("SOY-FREE", "SOY FREE", "NO SOY"));
+      put("no-eggs", Arrays.asList("EGG-FREE", "EGG FREE", "NO EGGS"));
+      put("no-alcohol", Arrays.asList("ALCOHOL-FREE", "ALCOHOL FREE", "NO ALCOHOL"));
+
+      // Question 3: What are your favorite types of cuisine?
+      put("african", Arrays.asList("AFRICAN", "NIGERIAN", "ETHIOPIAN", "EGYPTIAN", "CONGOLESE", "TANZANIAN", "SOUTH AFRICAN", "KENYAN", "UGANDAN", "ALGERIAN", "SUDANESE"));
+      put("american", Arrays.asList("AMERICAN"));
+      put("asian", Arrays.asList("ASIAN", "CHINESE", "INDIAN", "INDONESIAN", "PAKISTANI", "BANGLADESHI", "JAPANESE", "PHILIPINO", "FILIPINO", "VIETNAMESE", "KOREAN", "THAI"));
+      put("carribean", Arrays.asList("CARRIBEAN", "CUBAN", "HAITIAN", "DOMINICAN", "PUERTO RICAN", "JAMAICAN", "TRINIDADIAN", "COSTA RICAN"));
+      put("eastern-european", Arrays.asList("EASTERN EUROPEAN", "RUSSIAN", "UKRANIAN", "BELARUSIAN", "SLAVIC", "POLISH", "ROMANIAN", "MOLDOVAN", "SERBIAN", "SLOVAKIAN", "CROATIAN"));
+      put("western-european", Arrays.asList("WESTERN EUROPEAN", "GERMAN", "ENGLISH", "IRISH", "FRENCH", "SWISS", "BELGIAN", "DUTCH"));
+      put("latin-american", Arrays.asList("LATIN", "LATIN AMERICAN", "MEXICAN", "BRAZILIAN", "COLOMBIAN", "ARGENTINIAN", "PERUVIAN", "VENEZUELAN", "CHILEAN", "ECUADORIAN", "BOLIVIAN"));
+      put("mediterranean", Arrays.asList("MEDITERRANEAN", "GREEK", "ITALIAN", "SARDINIAN", "SPANISH"));
+      put("middle-eastern", Arrays.asList("MIDDLE EASTERN", "MIDDLE-EASTERN", "IRANIAN", "EGYPTIAN", "TURKISH", "IRAQI", "SAUDI ARABIAN", "ISRAELI", "JORDINIAN", "PALESTINIAN"));
+      put("south-asian", Arrays.asList("SOUTH ASIAN", "THAI", "CAMBODIAN", "MYANMA", "BURMESE", "INDONESIAN", "LAO", "LAOTIAN", "INDIAN"));
+
+      // Question 4: What are your general tastes?
+      put("spicy", Arrays.asList("SPICY"));
+      put("sweet", Arrays.asList("SWEET"));
+      put("salty", Arrays.asList("SALTY"));
+      put("crispy", Arrays.asList("CRISPY"));
+	  }
+  };
+
+  // Set the user's preferences
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    UserService userService = UserServiceFactory.getUserService();
+
+    // Retrieve the values of all checkboxes that the user did check in the quiz.
+    List<String> quizResponses = Arrays.asList(request.getParameterValues("choice"));
```

Comment:
oh i didnt realize it returned as a list thats so cool

Reply:
woah me neither

Reply:
To make this a bit faster, you can change quizResponses to be an array of strings (like String[]), so you don't have to go through the intermediate step of Arrays.asList